**Определение сердечно- сосудистых заболеваний Features: Возраст (в днях) Рост Вес Пол Систолическое кровяное давление Диастолическое кровяное давление Уровень холестерина 1: норма, 2: выше нормы, 3: сильно выше нормы Уровень глюкозы 1: норма, 2: выше нормы, 3: сильно выше нормы Курение Потребление алкоголя Физическая активность У - Наличие или отсутствие сердечно-сосудистых заболеваний (1 – есть заболевание, 0 – нет заболевания)**

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive/')

Drive already mounted at /content/gdrive/; to attempt to forcibly remount, call drive.mount("/content/gdrive/", force_remount=True).


In [ ]:
!ls -l '/content/gdrive/'

total 4
drwx------ 12 root root 4096 Jun  5 15:13 MyDrive


In [ ]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.decomposition import PCA
from sklearn.naive_bayes import GaussianNB
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn import metrics
from sklearn.pipeline import make_pipeline

In [ ]:
# Считываем данные из файла cardio.csv и загружаем их в датафрейм data
data = pd.read_csv('/content/gdrive/MyDrive/Colab Notebooks/ML/cardio_binary.csv')

In [ ]:
# Считываем данные из файла cardio.csv и загружаем их в датафрейм data
data = pd.read_csv('/content/gdrive/MyDrive/Colab Notebooks/ML/cardio_binary.csv', delimiter=';')
print(data)

          id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  18393       2     168    62.0    110     80            1     1   
1          1  20228       1     156    85.0    140     90            3     1   
2          2  18857       1     165    64.0    130     70            3     1   
3          3  17623       2     169    82.0    150    100            1     1   
4          4  17474       1     156    56.0    100     60            1     1   
...      ...    ...     ...     ...     ...    ...    ...          ...   ...   
69992  99993  19240       2     168    76.0    120     80            1     1   
69993  99995  22601       1     158   126.0    140     90            2     2   
69994  99996  19066       2     183   105.0    180     90            3     1   
69995  99998  22431       1     163    72.0    135     80            1     2   
69996  99999  20540       1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0 

In [ ]:
# Оставляем только необходимые признаки
features = ['id','age','gender','height','weight','ap_hi','ap_lo','cholesterol','gluc','smoke','alco','active','cardio']
data = data[features]
print(data)

          id    age  gender  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  18393       2     168    62.0    110     80            1     1   
1          1  20228       1     156    85.0    140     90            3     1   
2          2  18857       1     165    64.0    130     70            3     1   
3          3  17623       2     169    82.0    150    100            1     1   
4          4  17474       1     156    56.0    100     60            1     1   
...      ...    ...     ...     ...     ...    ...    ...          ...   ...   
69992  99993  19240       2     168    76.0    120     80            1     1   
69993  99995  22601       1     158   126.0    140     90            2     2   
69994  99996  19066       2     183   105.0    180     90            3     1   
69995  99998  22431       1     163    72.0    135     80            1     2   
69996  99999  20540       1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0 

In [ ]:
# Преобразуем категориальные признаки в бинарные
data = pd.get_dummies(data, columns=['gender', 'cholesterol', 'gluc', 'smoke', 'alco', 'active'], drop_first=True)


In [ ]:
# Отделяем признаки от целевой переменной
data_X = data.drop(columns=['cardio']).values
data_Y = data['cardio'].values

In [ ]:
# Разбиваем данные на обучающую и тестовую выборки в соотношении 70:30
X_train, X_test, y_train, y_test = train_test_split(data_X, data_Y, test_size=0.3, random_state=0)

In [ ]:
# Функция для вычисления метрик качества модели
def calc_metrics(true, pred):
    acc = metrics.accuracy_score(true, pred)
    auc = metrics.roc_auc_score(true, pred)
    f1 = metrics.f1_score(true, pred)

    print("Accuracy: {:.2%}\nROC AUC: {:.2%}\nF1: {:.2%}".format(acc, auc, f1))

In [ ]:
# Обучение и оценка Logistic Regression
lr_clf = make_pipeline(StandardScaler(), PCA(n_components=2), LogisticRegression(random_state=0))
lr_clf.fit(X_train, y_train)
pred_lr = lr_clf.predict(X_test)
print("Logistic Regression")
calc_metrics(y_test, pred_lr)


Logistic Regression
Accuracy: 59.34%
ROC AUC: 59.31%
F1: 49.55%


Полученные результаты относятся к модели логистической регрессии, которая была обучена и оценена на задаче определения сердечно-сосудистых заболеваний.

**Accuracy (Точность): 59.34%**

Это процент правильных предсказаний модели относительно общего числа примеров.
В данном случае, модель правильно классифицировала **около 59.34%** примеров тестовых данных.
ROC AUC (Площадь под кривой ROC): 59.30%

**ROC AUC** является метрикой, оценивающей качество бинарной классификации.
**Значение 59.31%** указывает на то, что модель имеет некоторую способность различать между положительными и отрицательными классами, но качество предсказания не является очень высоким.
F1-мера: 49.55%

**F1**-мера является сбалансированной метрикой, которая учитывает как точность, так и полноту модели.
**Значение 49.55%** указывает на среднюю гармоническую меру между точностью и полнотой модели.
Низкое значение F1-меры может говорить о том, что модель имеет проблемы с правильным и полным предсказанием классов.
Итак, результаты указывают на то, что модель логистической регрессии имеет некоторые ограничения в точности и способности различать между классами в задаче определения сердечно-сосудистых заболеваний. Возможно, есть место для улучшения модели или использования других алгоритмов машинного обучения для достижения более высоких показателей качества.